eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -338,6 +329,34 @@ jobs:
           parameters:
             ArtifactPath: $(Build.ArtifactStagingDirectory)/packages
             ArtifactName: packages
+        
+        - task: Powershell@2
+          inputs:
+            filePath: $(System.DefaultWorkingDirectory)/eng/common/scripts/Create-APIReview.ps1
+            arguments: >
+              -ArtifactList ('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object Name)
+              -OutPath $(Build.ArtifactStagingDirectory)
+              -ApiKey "$(azuresdk-apiview-apikey)"
+              -SourceBranch $(Build.SourceBranchName)
+              -DefaultBranch $(DefaultBranch)
+              -ParserPath $(System.DefaultWorkingDirectory)/parser/RelWithDebInfo/ParseAzureSdkCpp.exe
+              -ServicePath $(Build.SourcesDirectory)/sdk/${{ parameters.ServiceDirectory }}
+              -BuildId $(Build.BuildId)
+              -RepoName "$(Build.Repository.Name)"
+            pwsh: true
+            workingDirectory: $(Pipeline.Workspace)
+          displayName: Create API Review
+          condition: >-
+            and(
+              succeeded(),
+              ne(variables['Skip.CreateApiReview'], 'true'),
+              ne(variables['Build.Reason'],'PullRequest'),
+              eq(variables['System.TeamProject'], 'internal')
+            )
+        - template: /eng/common/pipelines/templates/steps/detect-api-changes.yml
```

Comment:
`Detect-API-changes` step should only run after publishing the artifacts. This step sends a request to APIView and APIView will try to access the published artifact when processing the request. If artifact is not published, then request will fail with 404 error.